### PR TITLE
add_new_dry_dirt

### PR DESCRIPTION
**Pull request by TumeniNodes**
New PR for dry dirt node. (ban the rebase command)
See original PR : https://github.com/minetest/minetest_game/pull/2409

### CODE REVIEW

**Comment by TumeniNodes** [comment]
I made no changes to the 'farming' mod this time, as the 'dirt_with_dry_grass' node is being left in the code, so it no longer throws out an error.
However, if it is felt that dry_dirt should be included, I will do so.
But, it seems problematic to be able to grow/farm in dry_dirt. It is a real life issue, and makes sense in MTG as well. (plus I'm lazy)

**Comment by Napiophelios** [comment]
how is any different than farming desert sand?

**Comment by TumeniNodes** [comment]
> how is any different than farming desert sand?

Can you farm in desert sand? o_0 I was not aware of this, as I never actually use/d 'farming'
That is completely bizarre IMO.
As I stated, _if_ it is agreed I should add the new node to farming, I will do so but maybe better to do in a separate PR?
The _only_ reason I made adjustments related to 'farming' is because removing the dirt_with_dry_grass node made it throw an error.
Leaving it in the code no longer causes the problem so I view that as another "request" for a separate PR. (don't judge me)

**Comment by TumeniNodes** [comment]
woops, forgot to add grass deco registrations for dry_dirt_with_dry_grass
I _knew_ something was missing. Looks much better now
screenshot_20190716_122806

**Comment by TumeniNodes** [comment]
@paramat this was a really good idea, it adds really nice colour variation in-game.
screenshot_20190716_124208

**Comment by paramat** [comment]
Yes looks good =)

All farmed dirt has to be watered first, so it seems reasonable to be able to farm savanna dirt. Also, the previous savanna dirt node could support farming, so the new one probably should.

Farming in desert sand was a mistake made years ago, that i would like to revert.

Sorry to mention this now, and i should have realied this earlier, but it's best that we simply keep 'dirt_with_dry_grass' named as it is and use this as the new dry dirt with dry grass node, the description will be 'Dry Dirt with ...'.
This way we don't have an unused node, and existing savannas will change their appearence.

**Comment by TumeniNodes** [comment]
> Yes looks good =)
> 
> All farmed dirt has to be watered first, so it seems reasonable to be able to farm savanna dirt. Also, the previous savanna dirt node could support farming, so the new one probably should.
> 
> Farming in desert sand was a mistake made years ago, that i would like to revert.


I will add to farming then.
And I fully agree regarding sand-farming

>Sorry to mention this now, and i should have realied this earlier, but it's best that we simply keep 'dirt_with_dry_grass' named as it is and use this as the new dry dirt with dry grass node, the description will be 'Dry Dirt with ...'.
>This way we don't have an unused node, and existing savannas will change their appearence.

This part I am confused about... 0_o
I did keep "dirt_with_dry_grass"
Are you saying I should simply switch the texture for "dirt_with_dry_grass" to be>
		{name = "default_dry_dirt.png^default_dry_grass_side.png", ? and then leave the rest as is,
aside from removing the "dry_dirt_with_dry_grass" node itself?

**Comment by TumeniNodes** [comment]
Actually, I'm just gonna go ahead and do what I listed above, and make all the changes as needed, where needed.
It actually makes more sense in the MTG way of things xD

**Comment by TumeniNodes** [comment]
changes made, and we have dry dirt farming
screenshot_20190716_135300

**Comment by paramat** [comment]
> Are you saying I should simply switch the texture for "dirt_with_dry_grass" to be>
{name = "default_dry_dirt.png^default_dry_grass_side.png", ?

Yes, you got it right. So we are simply changing the existing 'dirt with dry grass' node, and then adding a new 'dry dirt' node.

**Comment by TumeniNodes** [comment]
> > Are you saying I should simply switch the texture for "dirt_with_dry_grass" to be>
> > {name = "default_dry_dirt.png^default_dry_grass_side.png", ?
> 
> Yes, you got it right. So we are simply changing the existing 'dirt with dry grass' node, and then adding a new 'dry dirt' node.

Yep, already all done :)

**Comment by TumeniNodes** [comment]
There is now one problem.
In functions.lua / remove grass in darkness... I have been unable to work out how to have dry grass turn into  a dry dirt node, rather than a dirt node. I tried a couple ways but reverted as they failed.
screenshot_20190716_162908

**Comment by paramat** [comment]
Ok will think on that.

I just realised another problem.
By altering the appearance of 'dirt with dry grass', existing savanna will have that dry dirt appearance as the surface node, but will have normal 'dirt' below the surface.
So i think we will need a new 'dry dirt with dry grass' node after all. Keeping the 'dirt with dry grass' node too.

**Comment by TumeniNodes** [comment]
?? wha? I never even touched dungeon loot 0_o
    mods/dungeon_loot/mapgen.lua:56:8: variable 'biome' was previously defined on line 55

**Comment by TumeniNodes** [comment]
> Ok will think on that.
> 
> I just realised another problem, sorry i didn't realise before.
> By altering the appearance of 'dirt with dry grass', existing savanna will have that dry dirt appearance as the surface node, but will have normal 'dirt' below the surface.
> So i think we will need a new 'dry dirt with dry grass' node after all. Keeping the 'dirt with dry grass' node too.

That's what I had in the first place : /
And I have no idea wth travis is griping about, I never even touched 'dungeon loot'

**Comment by TumeniNodes** [comment]
nvm I see now... the latest merge sfan5 did tripped me up
bad sfan5 xD no supper for you

**Comment by paramat** [comment]
> Actually no, wait... I set the nodes underneath to be dry_dirt, did you look at all the changes yet?

Yes saw that, but that's only for newly generated savanna. Biome definitions only take effect for mapgen, not existing world.
Existing savanna still have 'dirt' nodes below the surface, so they will have to keep 'dirt with dry grass' on top.

If we end up using 'dry dirt with dry grass', the acacia tree. log and acacia bush will also need their 'on place' nodes updating.

**Comment by paramat** [comment]
Try:
```
--
-- Grass and dry grass removed in darkness
--

minetest.register_abm({
	label = "Grass covered",
	nodenames = {"group:spreading_dirt_type", "default:dry_dirt_with_dry_grass"},
	interval = 8,
	chance = 50,
	catch_up = false,
	action = function(pos, node)
		local above = {x = pos.x, y = pos.y + 1, z = pos.z}
		local name = minetest.get_node(above).name
		local nodedef = minetest.registered_nodes[name]
		if name ~= "ignore" and nodedef and not ((nodedef.sunlight_propagates or
				nodedef.paramtype == "light") and
				nodedef.liquidtype == "none") then
			if node.name == "default:dry_dirt_with_dry_grass" then
				minetest.set_node(pos, {name = "default:dry_dirt"})
			else
				minetest.set_node(pos, {name = "default:dirt"})
			end
		end
	end
})
```

File: mods/default/mapgen.lua [hunk 1]
```diff
@@ -576,8 +576,8 @@ function default.register_ores()
 		},
 		-- Only where default:dirt is present as surface material
 		biomes = {"taiga", "snowy_grassland", "grassland", "coniferous_forest",
-			"deciduous_forest", "deciduous_forest_shore", "savanna", "savanna_shore",
-			"rainforest", "rainforest_swamp", "floatland_coniferous_forest"}
+				"deciduous_forest", "deciduous_forest_shore", "rainforest",
+				"rainforest_swamp", "floatland_coniferous_forest"}
```
Comment (paramat) [inline mods/default/mapgen.lua]: ONly 1 indent needed for these 2 lines.

**Comment by TumeniNodes** [comment]
Ahhhhh, I tried that, but used elseif :P (I was _this_ close)

**Comment by TumeniNodes** [comment]
> > Actually no, wait... I set the nodes underneath to be dry_dirt, did you look at all the changes yet?
> 
> Yes saw that, but that's only for newly generated savanna. Biome definitions only take effect for mapgen, not existing world.
> Existing savanna still have 'dirt' nodes below the surface, so they will have to keep 'dirt with dry grass' on top.
> 
> If we end up using 'dry dirt with dry grass', the acacia tree. log and acacia bush will also need their 'on place' nodes updating.

now I am more confused, as I did that this last time with:
`      -- Savanna

	minetest.register_biome({
		name = "savanna",
		node_top = "default:dirt_with_dry_grass",
		depth_top = 1,
		node_filler = "default:dry_dirt",
		depth_filler = 1,
		node_riverbed = "default:sand",
		depth_riverbed = 2,
		node_dungeon = "default:cobble",
		node_dungeon_alt = "default:mossycobble",
		node_dungeon_stair = "stairs:stair_cobble",
		y_max = upper_limit,
		y_min = 1,
		heat_point = 89,
		humidity_point = 42,
	})

	minetest.register_biome({
		name = "savanna_shore",
		node_top = "default:dry_dirt",
		depth_top = 1,
		node_filler = "default:dry_dirt",
		depth_filler = 3,
		node_riverbed = "default:sand",
		depth_riverbed = 2,
		node_dungeon = "default:cobble",
		node_dungeon_alt = "default:mossycobble",
		node_dungeon_stair = "stairs:stair_cobble",
		y_max = 0,
		y_min = -1,
		heat_point = 89,
		humidity_point = 42,
	})

	minetest.register_biome({
		name = "savanna_ocean",
		node_top = "default:sand",
		depth_top = 1,
		node_filler = "default:sand",
		depth_filler = 3,
		node_riverbed = "default:sand",
		depth_riverbed = 2,
		node_cave_liquid = "default:water_source",
		node_dungeon = "default:cobble",
		node_dungeon_alt = "default:mossycobble",
		node_dungeon_stair = "stairs:stair_cobble",
		vertical_blend = 1,
		y_max = -2,
		y_min = -255,
		heat_point = 89,
		humidity_point = 42,
	})

	minetest.register_biome({
		name = "savanna_under",
		node_cave_liquid = {"default:water_source", "default:lava_source"},
		node_dungeon = "default:cobble",
		node_dungeon_alt = "default:mossycobble",
		node_dungeon_stair = "stairs:stair_cobble",
		y_max = -256,
		y_min = -31000,
		heat_point = 89,
		humidity_point = 42,
	})
`

**Comment by TumeniNodes** [comment]
(sigh) new PR https://github.com/minetest/minetest_game/pull/2415

**Comment by TumeniNodes** [comment]
need to close to redo since recent merge involving mapgen.lua

**Pull request closed without merging**